File: AWS_GUI_Dev/BAWGUI/Converters/VisibleIfNonZeroConverter.vb
Imports System.Globalization

Namespace Converters
    Public Class VisibleIfNonZeroConverter
        Implements IValueConverter

        Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.Convert
            Dim val = Double.Parse(value.ToString())
            If val = 0 Xor parameter = "HideIfNonZero" Then
                Return Visibility.Hidden
            Else
                Return Visibility.Visible
            End If
        End Function

        Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
            Throw New NotImplementedException()
        End Function
    End Class
End Namespace

File: AWS_GUI_Dev/BAWGUI.Results/Converters/VisibleIfNonZeroConverter.cs
using System;
using System.Collections.Generic;
using System.Globalization;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows;
using System.Windows.Data;

namespace BAWGUI.Results.Converters
{
    class VisibleIfNonZeroConverter:IValueConverter
    {
        object IValueConverter.Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            var val = Double.Parse(value.ToString());
            if (val == 0 ^ (parameter != null && parameter.ToString() == "HideIfNonZero"))
            {
                return Visibility.Hidden;
            }
            else
            {
                return Visibility.Visible;
            }
        }

        object IValueConverter.ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
        {
            return DependencyProperty.UnsetValue;
        }
    }
}
